File: java/src/main/java/mx/iteso/adapter/MSLicense.java
package mx.iteso.adapter;
/** MSLicense class. */
public class MSLicense {
    /**Supported string. */
    private String supported;
    /**Current version. */
    private float version;
    /**older version. */
    private static final float OLDER = 12.0f;
    /**newer version. */
    private static final float NEWER = 16.0f;
    /**Constructor.
     * @param nVersion version.
     */
    public MSLicense(final float nVersion) {
        this.version = nVersion;
        if (nVersion <= OLDER) {
            supported = "Old version, no support";
        } else if (nVersion > OLDER && nVersion <= NEWER) {
            supported = "Older version, still supported";
        } else if (nVersion > NEWER) {
            supported = "Current stable version";
        } else {
            supported = "Invalid License";
        }
    }
    /**
     * @return version.
     */
    public float getVersion() {
        return this.version;
    }
    /**
     * @return supported string.
     */
    public String isSupported() {
        return this.supported;
    }
}

File: python/src/main/python/MSLicense.py

class MSLicense():
    def __init__(self, version):
        self.version = version
        if nVersion <= 12.0:
            self.supported = "Old version, no support"
        elif nVersion > 12.0 and nVersion <= 16.0:
            self.supported = "Older version, still supported"
        elif (nVersion > 16.0):
            self.supported = "Current stable version"
        else :
            self.supported = "Invalid License"
    
    
    def getVersion(self):
        return self.version
    
    def isSupported(self):
        return self.supported
